### PR TITLE
BaseColumn::data_type()

### PR DESCRIPTION
**Pull request by mrzzzrm**
Makes it possible to resolve a column's derived type just from the `BaseColumn` object without the need for additional info, the DataType, that was before only known to the table.

### CODE REVIEW

File: src/lib/storage/chunk.hpp [hunk 1]
```diff
@@ -141,6 +141,16 @@ class Chunk : private Noncopyable {
    */
   size_t estimate_memory_usage() const;
 
+  /**
+   * @return A clone of this Chunk, with the same columns, allocator etc.
+   */
+  std::shared_ptr<Chunk> forward_clone() const;
```
Comment (mjendruk) [inline src/lib/storage/chunk.hpp]: This shouldn’t be part of the PR, should it?

File: src/lib/storage/reference_column.cpp [hunk 2]
```diff
@@ -12,7 +12,10 @@ namespace opossum {
 
 ReferenceColumn::ReferenceColumn(const std::shared_ptr<const Table> referenced_table,
                                  const ColumnID referenced_column_id, const std::shared_ptr<const PosList> pos)
-    : _referenced_table(referenced_table), _referenced_column_id(referenced_column_id), _pos_list(pos) {
+    : BaseColumn(referenced_table->column_data_type(referenced_column_id)),
```
Comment (mjendruk) [inline src/lib/storage/reference_column.cpp]: Cool

File: src/lib/CMakeLists.txt [hunk 3]
```diff
@@ -269,6 +270,7 @@ set(
     storage/dictionary_column/dictionary_encoder.hpp
     storage/dictionary_column.hpp
     storage/encoding_type.hpp
+    storage/materialize.hpp
```
Comment (mjendruk) [inline src/lib/CMakeLists.txt]: duplicate (see line 296)

**Comment by mrzzzrm** [comment]
Addressed

**Comment by mrzzzrm** [comment]
@mrks is this scale of PR (mostly refactoring, few loc) okay to be reviewed/merged without your review while you're away?

**Comment by mrks** [comment]
Yes, if you two are happy, I am too. Should you merge a PR before I have seen it, please send me an email. This is because I delete most of the Github Notifications but want to make sure that I don’t miss something.

**Pull request merged**